src/internal/operators/withLatestFrom.ts

```diff
@@ -1,107 +1,24 @@
 /** @prettier */
-import { ObservableInput, OperatorFunction, ObservedValueOf } from '../types';
+import { OperatorFunction, ObservableInputTuple, ObservedValueTupleFromArray, ObservableInput } from '../types';
 import { operate } from '../util/lift';
 import { OperatorSubscriber } from './OperatorSubscriber';
 import { innerFrom } from '../observable/from';
 import { identity } from '../util/identity';
 import { noop } from '../util/noop';
 import { popResultSelector } from '../util/args';
 
-/* tslint:disable:max-line-length */
-export function withLatestFrom<T, R>(project: (v1: T) => R): OperatorFunction<T, R>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>, R>(
-  source2: O2,
-  project: (v1: T, v2: ObservedValueOf<O2>) => R
-): OperatorFunction<T, R>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, R>(
-  v2: O2,
-  v3: O3,
-  project: (v1: T, v2: ObservedValueOf<O2>, v3: ObservedValueOf<O3>) => R
-): OperatorFunction<T, R>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>, R>(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  project: (v1: T, v2: ObservedValueOf<O2>, v3: ObservedValueOf<O3>, v4: ObservedValueOf<O4>) => R
-): OperatorFunction<T, R>;
-export function withLatestFrom<
-  T,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>,
-  R
->(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5,
-  project: (v1: T, v2: ObservedValueOf<O2>, v3: ObservedValueOf<O3>, v4: ObservedValueOf<O4>, v5: ObservedValueOf<O5>) => R
-): OperatorFunction<T, R>;
-export function withLatestFrom<
-  T,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>,
-  O6 extends ObservableInput<any>,
-  R
->(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5,
-  v6: O6,
-  project: (
-    v1: T,
-    v2: ObservedValueOf<O2>,
-    v3: ObservedValueOf<O3>,
-    v4: ObservedValueOf<O4>,
-    v5: ObservedValueOf<O5>,
-    v6: ObservedValueOf<O6>
-  ) => R
+export function withLatestFrom<T, O extends unknown[]>(inputs: [...ObservableInputTuple<O>]): OperatorFunction<T, [T, ...O]>;
+
+export function withLatestFrom<T, O extends unknown[]>(...inputs: [...ObservableInputTuple<O>]): OperatorFunction<T, [T, ...O]>;
+
+export function withLatestFrom<T, O extends unknown[], R>(
+  inputs: [...ObservableInputTuple<O>],
+  project: (...value: [T, ...O]) => R
```

Comment:
issue: The project needs to be moved into the same tuple that's used for the inputs - [as done here](https://github.com/ReactiveX/rxjs/blob/516166ed8dc389d57bf64dd536cb845dbb23402d/src/internal/operators/mergeWith.ts#L11-L14). It should look like this:

```ts
export function withLatestFrom<T, O extends unknown[], R>(
  inputsAndProject: [...ObservableInputTuple<O>, (...value: [T, ...O]) => R]
): OperatorFunction<T, R>;
```

Reply:
The test mentioned in the description works now correctly.